In [25]:
import numpy as np
import scipy.io as sp


La partie 'as [abréviation]' est optionnelle, elle permet d'abréger le nom de la bibliothèque lorsqu'on appelle l'une de ses fonctions. 

Exemple : au lieu d'écrire numpy.shape(), on peut maintenant écrire np.shape()

In [26]:
dataSet = sp.loadmat('breastw.mat') #On charge le dataset
print(type(dataset))

<class 'dict'>


la classe 'dict' désigne un type de donnée appelée dictionnaire: 
il met en relation des clés ('keys') avec des données à la manière d'un vrai dictionnaire suivant le modèle suivant :
                    'key' : 'value' ~= mot : définition
                    
                    
 Nos données sont stockées dans un dictionnaire, on doit à présent identifier les champs qui nous intéressent : les 'features' et les 'labels' de nos cellules

In [27]:
dataSet.keys()

dict_keys(['__header__', '__version__', '__globals__', 'X', 'y'])

Les clés ou 'keys' du dictionnaires étudié sont visibes ci-dessus. Il nous faut maintenat identifier à quelles clés sont associées nos données, on a en X les features et en y les labels si l'on suit la nomenclature classique.

In [32]:
X = dataSet['X'] #On extrait la donnée du champ 'X' du dictionnaire
np.shape(X) #shape() de la bibliothèque numpy (ici np), renvoie la taille de la matrice (array numpy) en argument


(683, 9)

La donnée 'X' est une matrice 683*9. Chaque ligne correspond à une cellule (il y en a donc 683), les 9 colonnes correspondent chacune à une feature.


In [29]:
label = dataSet['y'] #On extrait les données liées à la clé 'target'
print(np.shape(label))
print(label)

(683, 1)
[[0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 

On voit bien ici que le vecteur 'targets' correspond à nos labels (0 ou 1). Il y en a bien 569, le même nombre de cellules que dans 'data'.


In [33]:
malignIdx = np.argwhere(label == 1)#argwhere(condition) renvoie les indices (sous forme d'une liste) pour lesquels la condition est vérifiée
benignIdx = np.argwhere(label == 0)

Les deux lignes ci-dessus nous donnent les listes des indices de cellules correspondant à chaque classe (bénigne, Y = 0, ou maligne, Y = 1).

In [34]:
print(len(malignIdx)) #la fonction len() renvoie la longueur d'un vecteur
print(len(benignIdx))

239
444


On vérifie ici que les deux classes couvrent tous les points (683), c'est bien le cas.

A partir de ces listes d'indices, on peut maintenant séparer nos deux classes dans la matrice data : (travail en cours, j'updaterai bientôt)